src/clj/puppetlabs/services/ca/certificate_authority_service.clj

```diff
@@ -27,9 +30,9 @@
              comidi/routes->handler)
          settings
          path
-         wrap-with-authorization-check
-         puppet-version)))
-   context)
+         (fn [request] (wrap-with-authorization-check request {:oid-map oid-mappings}))
```

Comment:
I agree that it would be bit weird but I would probably prefer that weirdness over the current WETness.

I suppose another option would be to introduce a third service which rolled-up the OIDs from the CA and the middleware from the auth service.  Just spitballin' here.